include/cocaine/context.hpp

```diff
@@ -152,6 +169,9 @@ class context_t {
     // because services are allowed to start and stop other services during their lifetime.
     synchronized<service_list_t> m_services;
 
+    //Flag set after initialization
+    bool ready;
```

Comment:
In this case I can try to write some class similar to boost::signals (or even a wrapper on boost signals) which tracks all invocations and apply them to newly attached slots.
Or we can stop using signals and move that functions to basic_dispatch interface. In this case usage will be limited only by basic_dispatch derived classes. I don't really like both of them - maybe you have a better idea?